### PR TITLE
"Lazy" property in Service annotation: Define service as lazy

### PR DESCRIPTION
**Pull request by alber999**
Wouldn't it be nice to define lazy services in Service annotation?

```
/**
 * @Service("my_service", lazy=true)
 */
class MyService { ... }
```

Just by adding "ocramius/proxy-manager" as dependency and handling Service "lazy" property you can make it. I hope you find this pull request useful

Cheers

### CODE REVIEW

**Comment by alber999** [comment]
This tests will always fail:
https://travis-ci.org/schmittjoh/JMSDiExtraBundle/jobs/93056101

traits where added in PHP 5.4 but tests run on PHP 5.3.29

All other tests pass ok (except allowed failures)

**Comment by alber999** [comment]
This works perfectly in full-stack symfony framewrok, but in standalone mode one more dependency should be added: "symfony/proxy-manager-bridge"

**Comment by TomasVotruba** [comment]
@alber999 Nice work!

As for tests, you can mark trait tests skipped for PHP 5.3-

**Comment by wodka** [comment]
@alber999 can you rebase your branch? this was fixed in the TraitTest

**Comment by alber999** [comment]
@wodka done!

File: Metadata/ClassMetadata.php [hunk 1]
```diff
@@ -52,6 +71,7 @@ public function serialize()
             $this->scope,
             $this->public,
             $this->abstract,
+            $this->lazy,
```
Comment (Ener-Getick) [inline Metadata/ClassMetadata.php]: It should be the last one to avoid BC breaks.

File: Metadata/ClassMetadata.php [hunk 2]
```diff
@@ -60,30 +80,42 @@ public function serialize()
...
 
-        list(
+        // prevent errors if not all key's are set
+        @list(
             $this->id,
             $this->parent,
             $this->scope,
             $this->public,
             $this->abstract,
+            $this->lazy,
```
Comment (Ener-Getick) [inline Metadata/ClassMetadata.php]: same thing here

File: composer.json [hunk 3]
```diff
@@ -19,7 +19,8 @@
         "symfony/process": "~2.3|~3.0",
         "symfony/finder": "~2.3|~3.0",
         "jms/aop-bundle": "~1.1",
-        "jms/metadata": "~1.0"
+        "jms/metadata": "~1.0",
+        "ocramius/proxy-manager": "~1.0"
```
Comment (Ener-Getick) [inline composer.json]: Why do we need this ?

**Comment by Ener-Getick** [comment]
Well I think I'll merge https://github.com/schmittjoh/JMSDiExtraBundle/pull/145 instead.
So I close this PR in favor of https://github.com/schmittjoh/JMSDiExtraBundle/pull/145.

**Pull request closed without merging**